leftwm-core/src/handlers/display_event_handler.rs

```diff
@@ -20,6 +20,7 @@ impl<C: Config, SERVER: DisplayServer> Manager<C, SERVER> {
                 .mouse_combo_handler(mod_mask, button, handle, x, y),
 
             DisplayEvent::WindowTakeFocus(handle) => from_window_take_focus(state, handle),
+            DisplayEvent::WindowSwitchToTag(handle) => from_window_switch_to_tag(state, handle),
             DisplayEvent::HandleWindowFocus(handle) => from_handle_window_focus(state, handle),
```

Comment:
You can just focus the window using `WindowTakeFocus`, there's no need to add a new event type.

Also, your event technically does not focus the window but only the tag it is on (if there's another window on that tag and the mouse hovers over it, it gets focused instead of the browser).

Reply:
See my other comment about this not working. I'm going to try again to see if I can figure out what's going wrong with `WindowTakeFocus`.

Reply:
I think I found the bug. I'll push the fix as a separate patch with an explanation.

Reply:
Found the bug, I've got a partial fix but working on resolving one more issue. I'll push as a separate commit.